clang/test/Modules/non-affecting-input-file-rebuilds.c

```diff
@@ -0,0 +1,61 @@
+// This test checks that reading a non-affecting module map during PCM
+// compilation doesn't cause the PCM to be recompiled when the file changes.
+
+// RUN: rm -rf %t && mkdir %t
+// RUN: split-file %s %t
+
+// XFAIL: *
+// rdar://101268970
```

Comment:
Did you still want to include this test in the patch? Why does it currently fail?

Reply:
I did, because I believe this much better checks for what the original patch https://reviews.llvm.org/D106876 intended to achieve. It fails because different PCMs are produced when compiling module B depending whether the non-affecting module map for module A is or is not on the filesystem. With my patch, signature of B.pcm now changes (since we serialize non-affecting input files into the PCM again). Changing the signature of B causes Clang to recompile module C, which is not desirable. Everything should remain stable/unchanged when only non-affecting files change.